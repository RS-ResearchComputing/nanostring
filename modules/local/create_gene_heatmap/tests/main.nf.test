nextflow_process {

    name "Test Process CREATE_GENE_HEATMAP"
    script "../main.nf"
    process "CREATE_GENE_HEATMAP"

    tag "modules"
    tag "modules_nfcore"
    tag "CREATE_GENE_HEATMAP"

    test("test profile") {

        when {
            process {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'nanostring/modules_test_data/test_profile/counts_Norm_GEX_ENDO.tsv', checkIfExists: true )
                input[1] = file(params.pipelines_testdata_base_path + 'nanostring/modules_test_data/test_profile/normalized_counts.tsv', checkIfExists: true )
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success},
                { assert snapshot(
                    process.out.gene_heatmap.collect { file(it).name }, //list unstable png files
                    process.out.versions
                    ).match()
                }
            )
        }
    }
}
