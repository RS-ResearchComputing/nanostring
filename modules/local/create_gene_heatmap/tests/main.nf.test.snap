{
    "test profile": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "gene_heatmap": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-19T20:27:50.118882334"
    }
}